usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -9318,59 +9302,199 @@ CK_RV ep11tok_get_mechanism_info(STDLL_TokData_t * tokdata,
     return tokdata->policy->update_mech_info(tokdata->policy, type, pInfo);
 }
 
+static CK_RV ep11_config_add_apqn(ep11_private_data_t *ep11_data,
+                                  struct ConfigNumPairNode *pair,
+                                  const char *fname)
+{
+    if (pair->value1 > 255) {
+        OCK_SYSLOG(LOG_ERR, "%s: Error: Expected valid adapter"
+                   " number, found '%lu' in config file '%s' at line %d\n",
+                   __func__, pair->value1, fname, pair->base.line);
+        TRACE_ERROR(" Error: Expected valid adapter number, found '%lu' in "
+                    "config file '%s' at line %d\n", pair->value1, fname,
+                    pair->base.line);
+        return CKR_ARGUMENTS_BAD;
+    }
 
-/* used for reading in the adapter config file,
- * converts a 'token' to a number, returns 0 with success
- */
-static inline short check_n(ep11_target_t * target, char *nptr, int *apqn_i)
+    if (pair->value2 > 255) {
+        OCK_SYSLOG(LOG_ERR, "%s: Error: Expected valid domain"
+                   " number, found '%lu' in config file '%s' at line %d\n",
+                   __func__, pair->value2, fname, pair->base.line);
+        TRACE_ERROR(" Error: Expected valid domain number, found '%lu' in "
+                    "config file '%s' at line %d\n", pair->value2, fname,
+                    pair->base.line);
+        return CKR_ARGUMENTS_BAD;
+    }
+
+    if (ep11_data->target_list.length >= MAX_APQN) {
+        OCK_SYSLOG(LOG_ERR,"%s: Error: Too many APQNs in config "
```

Comment:
I can't judge from just reading this code snippet: is it intentionally '>= MAX_APQN' ? The old code checks for '> MAX_APQN'

Reply:
Yes I would say >= is correct (and > was wrong previously). When the length is already equal to MAX_APQN then the list is already full.